### PR TITLE
BUGFIX: Loading Scalar Param from `.dat` file

### LINKED ISSUE #120: DataPortal bug
# bug found by: Maria Marcos Nuñez
```python
from pyomo.environ import *
model = AbstractModel()

# Definición de los diferentes parametros del modelo
model.qv = Param(within = NonNegativeReals)

# Definicion de las variables
model.D = Var(within = NonNegativeReals, bounds =(0.2, 6),initialize = 3, doc='Diametro del reactor')

# silly objective function supplied by DLW
def _e(m):
    return m.qv * m.D
model.silliness = Objective(rule=_e)

data = DataPortal()
data.load(filename="foobar.dat")
mc = model.create_instance(data)
```

# contents of foobar.dat:
```
param qv := 7.734;
```

# output (after stack trace):
```
ValueError: Parameter 'qv' defined with '1' dimensions, but data has '1' values: [7.734]. Are you missing a value for a 1-dimensional index?
```

### PR DESCRIPTION
**Pull request by marvin-meck**
## Summary/Motivation:

Loading scalar `param` from `.dat` files failes when not providing a `model` to the `load` method. 
The function `_process_data_list` from module `pyomo/dataportal/process_data.py` is called with `dim=1` instead of `dim=0`. 
Thus an indexed parameter is expected instead of a scalar one. 
This PR adds a check to pass the correct value for `dim`

## Changes proposed in this PR:
- `pyomo/dataportal/process_data.py`  line 346: added `_dim = 1 if len(cmd) > 1 else 0` to check for the correct value of dim.
- `pyomo/dataportal/process_data.py`  line 347: changed to `finaldata = _process_data_list(pname, _dim, cmd)` from `finaldata = _process_data_list(pname, 1, cmd)` 
- a unit test `test_md18` was added to `pyomo/blob/bugfix-no120/pyomo/dataportal/tests/test_dataportal.py` along with a data file `https://github.com/marvin-meck/pyomo/blob/bugfix-no120/pyomo/dataportal/tests/data17.dat`

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by mrmundt — APPROVED** [review]

**Comment by mrmundt** [comment]
Thanks, @marvin-meck - this looks good to me. We have a policy that two core devs need to review contributions, so once we get a second review in, we'll merge this. We appreciate you taking the time to do this!

**Review by jsiirola — APPROVED** [review]
Looks good!

**Pull request merged**
